src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/Listeners/StorageAnalyticsLogParser.cs

```diff
@@ -80,7 +85,12 @@ public StorageAnalyticsLogParser(ILogger<BlobListener> logger)
         {
             List<StorageAnalyticsLogEntry> entries = new List<StorageAnalyticsLogEntry>();
 
-            using (TextReader tr = new StreamReader(await blob.OpenReadAsync(cancellationToken)))
+            OperationContext operationContext = new OperationContext { ClientRequestID = Guid.NewGuid().ToString() };
+            using (TextReader tr = new StreamReader(await TimeoutHandler.ExecuteWithTimeout("ParseLogAsync", operationContext.ClientRequestID, _exceptionHandler,
+                    _logger, cancellationToken, () =>
+                    {
+                        return blob.OpenReadAsync(accessCondition: null, options: null, operationContext: operationContext, cancellationToken);
```

Comment:
Not clear to me that the OpenReadAsync API could possibly be affected by the original issue addressed by https://github.com/Azure/azure-storage-net/pull/985 - @brettsam?

Reply:
The hangs were almost all a result of the SDK internally parsing XML responses from the storage apis like ListBlobs (if i recall). You can see the stack in the linked issue above.

This operation directly on a blob doesn't look to be in the same family as those -- but it seems like we need to start gathering more details here to figure out why this is stopping.